compiler/src/iree/compiler/API/Internal/IREEGPUDialectCAPI.cpp

```diff
@@ -420,5 +424,119 @@ ireeHALExecutableTargetAttrGetGPUTargetInfo(MlirAttribute attr) {
       wgpAttr.getMaxThreadCountPerWorkgroup();
   targetInfo.maxWorkgroupMemoryBytes = wgpAttr.getMaxWorkgroupMemoryBytes();
 
+  targetInfo.mmaIntrinsics = wrap(builder.getArrayAttr({}));
+  mlir::iree_compiler::IREE::GPU::MMAOpsArrayAttr mmaOpsArray =
+      wgpAttr.getMma();
+  if (!mmaOpsArray) {
+    return targetInfo;
+  }
+
+  std::vector<mlir::Attribute> mmaIntrinsicAttrs;
+  for (mlir::iree_compiler::IREE::GPU::MMAAttr mmaAttr : mmaOpsArray) {
+    mlir::iree_compiler::IREE::GPU::MMAIntrinsic intrinsic =
+        mmaAttr.getIntrinsic();
+    mmaIntrinsicAttrs.push_back(
+        mlir::iree_compiler::IREE::GPU::MMAIntrinsicAttr::get(context,
+                                                              intrinsic));
+
+    for (mlir::iree_compiler::IREE::GPU::VirtualMMAIntrinsic virtualIntrinsic :
+         mmaAttr.getVirtualIntrinsics()) {
+      mmaIntrinsicAttrs.push_back(
+          mlir::iree_compiler::IREE::GPU::VirtualMMAIntrinsicAttr::get(
+              context, virtualIntrinsic));
+    }
+  }
+  targetInfo.mmaIntrinsics = wrap(builder.getArrayAttr(mmaIntrinsicAttrs));
+  return targetInfo;
+}
+
+ireeGPUTargetInfo ireeGPUTargetInfoGet(
+    MlirContext mlirCtx, const char *arch, const int32_t *subgroupChoices,
+    size_t numSubgroupChoices, const int32_t *workgroupSizes,
+    size_t numWorkgroupSizes, int32_t threadCount, int32_t memoryBytes,
+    const mma_intrinsic_enum_t *mmaIntrinsics, size_t numMmaIntrinsics) {
+  assert(!mlirContextIsNull(mlirCtx) && "mlirCtx cannot be null");
+  assert(arch && "arch cannot be null");
+
+  mlir::MLIRContext *context = unwrap(mlirCtx);
+  mlir::Builder builder(context);
+
+  ireeGPUTargetInfo targetInfo = {};
+
+  targetInfo.arch = wrap(mlir::StringAttr::get(context, arch));
+  std::vector<int32_t> subgroupChoicesVec(subgroupChoices,
+                                          subgroupChoices + numSubgroupChoices);
+  targetInfo.subgroupSizeChoices =
+      wrap(builder.getI32ArrayAttr(subgroupChoicesVec));
+  std::vector<int32_t> workgroupSizesVec(workgroupSizes,
+                                         workgroupSizes + numWorkgroupSizes);
+  targetInfo.maxWorkgroupSizes =
+      wrap(builder.getI32ArrayAttr(workgroupSizesVec));
+
+  targetInfo.maxThreadCountPerWorkgroup = threadCount;
+  targetInfo.maxWorkgroupMemoryBytes = memoryBytes;
+
+  std::vector<mlir::Attribute> mmaIntrinsicAttrs;
+  mmaIntrinsicAttrs.reserve(numMmaIntrinsics);
+  for (size_t i = 0; i < numMmaIntrinsics; i++) {
+    mma_intrinsic_enum_t enumValue = mmaIntrinsics[i];
```

Comment:
Can we use a range for loop here? I don't see `i` being used anywhere else.

Also https://llvm.org/docs/CodingStandards.html#prefer-preincrement